Framework/Parallel/test/EventLoaderTest.h

```diff
@@ -0,0 +1,116 @@
+  }
+
+  const std::vector<int64_t> &eventIndex() const override { return m_index; }
+  const std::vector<int64_t> &eventTimeZero() const override {
+    return m_time_zero;
+  }
+  void readEventID(int32_t *event_id, size_t start,
+                   size_t count) const override {
+    for (size_t i = 0; i < count; ++i)
+      event_id[i] = static_cast<int32_t>(m_bank * m_bankSize +
+                                         (start + count) % m_bankSize);
+  }
+  void readEventTimeOffset(int32_t *event_time_offset, size_t start,
+                           size_t count) const override {
+    for (size_t i = 0; i < count; ++i)
+      event_time_offset[i] = static_cast<int32_t>(start + count);
+  }
+
+private:
+  const size_t m_bankSize{10};
+  size_t m_bank;
+  std::vector<int64_t> m_index;
+  std::vector<int64_t> m_time_zero;
+};
+}
+
+class EventLoaderTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static EventLoaderTest *createSuite() { return new EventLoaderTest(); }
+  static void destroySuite(EventLoaderTest *suite) { delete suite; }
+
+  void test_throws_if_file_does_not_exist() {
+    TS_ASSERT_THROWS(EventLoader::load("abcdefg", "", {}, {}, {}),
+                     H5::FileIException);
+  }
+
+  void test_H5DataType_parameter_pack_conversion() {
+    using EventLoader::load;
+    TS_ASSERT_THROWS_EQUALS(load(H5::PredType::NATIVE_INT32),
+                            const std::runtime_error &e, std::string(e.what()),
+                            "int32_t");
+    TS_ASSERT_THROWS_EQUALS(load(H5::PredType::NATIVE_INT64),
+                            const std::runtime_error &e, std::string(e.what()),
+                            "int64_t");
+    TS_ASSERT_THROWS_EQUALS(load(H5::PredType::NATIVE_UINT32),
+                            const std::runtime_error &e, std::string(e.what()),
+                            "uint32_t");
+    TS_ASSERT_THROWS_EQUALS(load(H5::PredType::NATIVE_UINT64),
+                            const std::runtime_error &e, std::string(e.what()),
+                            "uint64_t");
+    TS_ASSERT_THROWS_EQUALS(load(H5::PredType::NATIVE_FLOAT),
+                            const std::runtime_error &e, std::string(e.what()),
+                            "float");
+    TS_ASSERT_THROWS_EQUALS(load(H5::PredType::NATIVE_DOUBLE),
+                            const std::runtime_error &e, std::string(e.what()),
+                            "double");
+    TS_ASSERT_THROWS_EQUALS(
+        load(H5::PredType::NATIVE_CHAR), const std::runtime_error &e,
+        std::string(e.what()),
+        "Unsupported H5::DataType for entry in NXevent_data");
+  }
+
+  void test_load() {
+    const std::vector<size_t> bankSizes{10, 100, 1000};
+    const size_t chunkSize{37};
+    Chunker chunker(1, 0, bankSizes, chunkSize);
+    ::detail::FakeDataSource dataSource;
```

Comment:
👍  for arranging load methods in such a way as not to need file for  unittesting of this.